### PR TITLE
Make userID invisible from AppStore builds

### PR DESCRIPTION
**Pull request by alizreszlerlumo**
https://lumoslabs.atlassian.net/browse/IOS-795

### CODE REVIEW

File: src/ofxGenericSettings.cpp [hunk 1]
```diff
@@ -29,7 +31,10 @@ bool ofxGenericSettings::readFromDisk()
         parse( oldParse );
     } else
     {
-        ofxGLogError( "Unable to read settings file " + getFileName() );
+        // We should never log paths containing the users id to the console or 3rd party logging systems.
+        NSString *objCFileName = ofxStringToNSString(getFileName());
+        NSString *lastPathComponent = [objCFileName lastPathComponent];
+        ofxGLogError( "Unable to read settings file " + ofxNSStringToString(lastPathComponent) );
```
Comment (jessenaugher) [inline src/ofxGenericSettings.cpp]: can lastPathComponent be nil here?

File: src/ofxGenericUtility.cpp [hunk 2]
```diff
@@ -195,6 +195,15 @@ void ofxGLogWarning( const string & message )
 	// IOSP-40 // ofLogWarning( ofxGenericModuleName, message );
 }
 
+void ofxGLogDebug( const string & message )
+{
+    ofPtr< ofxGenericApp > app = ofxGenericApp::getInstance();
+    if (app) {
+        app->ofxLLogDebug(message);
+    }
+    // IOSP-40 // ofLogWarning( ofxGenericModuleName, message );
```
Comment (jessenaugher) [inline src/ofxGenericUtility.cpp]: do we need this?

File: src/ofxGenericValueStore.cpp [hunk 3]
```diff
@@ -1359,6 +1360,11 @@ void ofxGenericValueStore::setFileName( string fileName, bool fileInDocuments )
 {
     _fileName = fileName;
     _fileInDocuments = fileInDocuments;
+    
+    // We should never log paths containing the users id to the console or 3rd party logging systems.
+    NSString *objCFileName = ofxStringToNSString(_fileName);
+    NSString *lastPathComponent = [objCFileName lastPathComponent];
+    _lastPathComponentInFileName = ofxNSStringToString(lastPathComponent);
```
Comment (jessenaugher) [inline src/ofxGenericValueStore.cpp]: same question

File: src/ofxGenericValueStore.cpp [hunk 4]
```diff
@@ -1378,7 +1384,8 @@ bool ofxGenericValueStore::readFromDisk()
             
             if( _verify && !verifyContentsFromDisk() )
             {
-                ofxGLogError("ofxGenericValueStore::readFromDisk file " + _fileName + " failed on verification!");
+                // TODO: this filename can contain the users ID, we should log only the actual filename.
```
Comment (jessenaugher) [inline src/ofxGenericValueStore.cpp]: this todo can go away right? or no?

File: src/ofxGenericSettings.cpp (hunk 1 above)
Comment (alizreszlerlumo) [inline src/ofxGenericSettings.cpp]: only if the path is nil ... lastPathComponent should always return the string if it doesn't have any paths in it.

File: src/ofxGenericValueStore.cpp (hunk 3 above)
Comment (alizreszlerlumo) [inline src/ofxGenericValueStore.cpp]: Adde some extra measures to avoid nil here.

**Pull request merged**